mern/client/src/screens/RecoverSession/recoverSession.js

```diff
@@ -1,105 +1,123 @@
-import React, { Component } from "react";
+import React, { useState, useContext } from "react";
 import "./recoverSession.css";
 import GenericButton from "../../Components/Buttons/GenericButton";
-import { withRouter } from "../withRouter";
-class RecoverSession extends Component {
-  constructor(props) {
-    super(props);
+import { NavStateContext } from "../../Components/Navbar/navState";
+import { useNavigate } from "react-router-dom";
 
-    this.state = { value: "" };
-    this.handleChange = this.handleChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
```

Comment:
This is a good catch. We need to change the remaining class based files and change them into functional React. Keep doing this as you find them. Also made an issue about this